haystack/components/samplers/top_p.py

```diff
@@ -51,20 +56,28 @@ def __init__(self, top_p: float = 1.0, score_field: Optional[str] = None):
     @component.output_types(documents=List[Document])
     def run(self, documents: List[Document], top_p: Optional[float] = None):
         """
-        Filter documents based on their similarity scores using top-p sampling.
+        Filters documents using top-p sampling based on their scores.
+
+        :param documents: List of Document objects to be filtered.
+        :param top_p: Optional. A float to override the cumulative probability threshold set during initialization.
+                 If None, the class's top_p value is used.
+        :return: A dictionary with a key 'documents' containing the list of filtered Document objects.
+
+        This method applies top-p sampling to filter out documents. It selects those documents whose similarity scores
+        are within the top 'p' percent of the cumulative distribution, based on the specified or default top_p value.
+
+        If the specified top_p results in no documents being selected (especially in cases of a low top_p value), the
+        method defaults to returning the document with the highest similarity score.
```

Comment:
is it ok to move this section above param descriptions? We usually have it in the beginning to explain the method.